docs/how-to/uc-image-creation.rst

```diff
@@ -5,11 +5,141 @@ How to create a real-time Ubuntu Core image
 
     This guide assumes access to features available exclusively to `dedicated Snap Store`_ users.
 
-The :doc:`../how-to/uc-boot-parameters` describes the method for dynamically configuring the system for real-time processing.
-The configuration is an iterative process that is best done together with the expected workload. 
-Once satisfied with the configurations, it is time to prepare the operating system for going into production.
+Canonical builds and publishes Ubuntu Core images for a range of `supported platforms`_.
+However, these images include the generic Linux kernel. 
+In order to run Ubuntu Core with the real-time kernel, we need to build it ourselves. 
 
-This guide shows how to statically set the desired Kernel command-line options for the Ubuntu Core system.
+This guide shows how to build and Ubuntu Core image with the real-time kernel.
+We start by showing how to :ref:`build a vanilla image <ubuntu-core-image>` which is excellent for testing and tuning, and then illustrate how to :ref:`make a custom image <custom-ubuntu-core-image>` with production-ready kernel configurations.
+
+
+.. _ubuntu-core-image:
+
+Create an image with the real-time kernel
+-----------------------------------------
+
+To do this we need to describe the image content and then use a tool to build it.
+
+.. _model-assertion:
+
+Create the model assertion
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The model assertion is a digitally signed document that describes the content of the Ubuntu Core image.
+Read the `model assertion`_ documentation before continuing.
+
+Below is an example model assertion in YAML, describing a `core22` Ubuntu Core
+image:
+
+.. literalinclude :: uc-image-creation/model.json
+   :language: json
+
+Inside an empty directory, create a file named ``model.json`` with the above content.
+
+Change the following:
+
+- ``authority-id``, ``brand-id`` to your developer ID, since this is custom model. Use ``snapcraft whoami`` command to get your developer ID.
+- ``timestamp`` to a RFC3339 formatted string, within you signing key's validity. Use ``date -Iseconds --utc`` command to use the current time.
+- ``store`` to your dedicated Snap Store ID.
+- ``model`` to something representative of your model.
+
+The ``snaps`` array is a list of snaps that get included in the image.
+In that list, the ``realtime-kernel`` snap contains the realtime Linux kernel.
+Here you can add any other snaps, including for example your real-time applications.
+
+Next, we need to sign the model assertion.
+Refer to the guide on `signing model assertion`_ for details on how to sign the model assertion. 
+
+Here are the needed steps:
+
+1) Create and register a key
+
+
+.. code-block:: shell
+
+    snapcraft create-key realtime-ubuntu
+    snapcraft register-key realtime-ubuntu
```

Comment:
Regarding this, it's important to advise the user to choose the key **name** and **password** carefully because once registered, [it's not possible to remove the key from the Snapcraft store](https://forum.snapcraft.io/t/remove-registered-signing-key-from-snap-store/27280).

Consider adding a warning in this section.

Reply:
Not sure why they need to choose the name and password carefully. Based on the explanation on the forum post, the registered key is the public part and can't be removed.

Reply:
Let me put this differently. It’s not so much about being careful, but rather being **aware** of the implications of creating a key. This process is **permanent** and **irreversible**, and anything with those characteristics should come with a warning to the user. 

Additionally, the user cannot delete a key even if it isn't registered yet. Since creating a key requires performing `snapcraft login` first, the key remains tied to the user account regardless.

The key reasons to warn the user are:

1. **Irreversibility**: Permanent actions, by definition, cannot be undone. The user may not realize they are performing an irreversible action.

2. **Security Implications**: The user is creating a key with a password, and they won’t be able to change the password after the key is created and registered.

3. **Avoiding Future Frustration**: The user could become frustrated if, after following the documentation, they wish to clean up everything and discover they can't.

This warning is important to prevent user mistakes, provide a better user experience, and ensure clarity when following the documentation.

Reply:
> Additionally, the user cannot delete a key even if it isn't registered yet. Since creating a key requires performing snapcraft login first, the key remains tied to the user account regardless.

I don't think that's true. The key is stored locally. Only registered public keys are uploaded and tied to an Ubuntu account.

Please provide your feedback to snapcraft and core documentations as they are out of the scope of this document. As mentioned, the user is asked to read the documentation before starting.